src/analysis/convex/cone.lean

```diff
@@ -643,4 +643,23 @@ begin
       set.mem_singleton_iff, forall_eq, imp_self] }
 end
 
+lemma closed_inner_dual_cone : is_closed (s.inner_dual_cone : set H) :=
+begin
+  -- reduce the problem to showing that dual cone of a singleton `{x}` is closed
+  rw inner_dual_cone_eq_Inter_inner_dual_cone_singleton,
+  apply is_closed_Inter,
+  simp_rw [set_coe.forall, subtype.coe_mk],
+  intros x hx,
+
+  -- the dual cone of a singleton set is the preimage of `[0, ∞)` under `inner x`
+  have h : (({x} : set H).inner_dual_cone : set H) = (inner x : H → ℝ)⁻¹' (set.Ici 0),
+  { apply set.ext,
+    simp only [convex_cone.mem_coe, mem_inner_dual_cone, mem_singleton_iff, forall_eq, mem_preimage,
+    mem_Ici, iff_self, forall_const] },
```

Comment:
If all of these are `simp` lemmas, you can replace everything by `simp`, because it is terminal, and then your proof can be golfed to `by { ext, simp }` or maybe even `by simp`.

Reply:
Sounds good, thanks! I got this list from `squeeze_simp` :D

Reply:
`simp` needed to be passed `mem_inner_dual_cone`. This has happened in a previous lemma as well so I just added `@[simp]` to that lemma.